torch/_functorch/aot_autograd.py

```diff
@@ -1062,7 +1144,37 @@ def aot_dispatch_base(flat_fn, flat_args: List[Tensor], aot_config: AOTConfig):
 
     @wraps(compiled_fw)
     def new_fn(args):
-        fw_outs = call_func_with_args(compiled_fw, args, disable_amp=disable_amp)
+        if _synthetic_base_info is not None:
+            new_inputs, metadata = merge_view_inputs(args, _input_info)
+            args_with_synthetic_bases = new_inputs
+        else:
+            args_with_synthetic_bases = args
+        updated_inputs, fw_outs = call_func_with_args(compiled_fw, args_with_synthetic_bases, disable_amp=disable_amp)
+
+        assert len(updated_inputs) == _num_mutated_inputs
+        assert len(fw_outs) == _num_outputs
+
+        if _num_mutated_inputs > 0:
```

Comment:
this is pretty close to what `CompiledFunction.apply` wrapper is doing, are there some helper functions that can be refactored out?